packages/meditrak-app/app/assessment/selectors.js

```diff
@@ -109,3 +110,85 @@ export const getQuestion = (state, questionId) => state.assessment.questions[que
 export const getAnswerForQuestion = (state, questionId) => getAnswers(state)[questionId];
+
+export const getArithmeticResult = (state, arithmeticQuestionId) => {
+  const { config } = getQuestion(state, arithmeticQuestionId);
+
+  const {
+    arithmetic: { formula, valueTranslation = {}, defaultValues = {}, answerDisplayText = '' },
+  } = config;
+
+  const values = {};
+  const expressionParser = new ExpressionParser();
+  const variables = expressionParser.getVariables(formula);
+  let translatedAnswerDisplayText = answerDisplayText;
+
+  const getArithmeticAnswer = questionId => {
+    const answer = getAnswerForQuestion(state, questionId);
+
+    if (valueTranslation[questionId] && valueTranslation[questionId][answer] !== undefined) {
+      return valueTranslation[questionId][answer]; // return translated answer if there's any
+    }
+
+    if (answer !== undefined) {
+      return answer; // return raw answer
+    }
+
+    return defaultValues[questionId] !== undefined ? defaultValues[questionId] : 0; // No answer found, return the default answer
+  };
+
+  // Setting up scope values.
+  variables.forEach(questionIdVariable => {
+    const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix to get the actual questionId
+    values[questionIdVariable] = getArithmeticAnswer(questionId);
+  });
+
+  // Replace variables with real values in answerDisplayText
+  variables.forEach(questionIdVariable => {
+    const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix to get the actual questionId
+    const answer = getArithmeticAnswer(questionId);
+    translatedAnswerDisplayText = translatedAnswerDisplayText.replace(questionId, answer);
+  });
+
+  expressionParser.setScope(values);
+
+  const result = !isNaN(expressionParser.evaluate(formula))
+    ? Math.round(expressionParser.evaluate(formula) * 1000) / 1000 // Round to 3 decimal places
+    : 0;
+  translatedAnswerDisplayText = translatedAnswerDisplayText.replace('$result', result);
+
+  return {
+    answerDisplayText: translatedAnswerDisplayText,
+    result,
+  };
+};
+
+export const getConditionResult = (state, conditionQuestionId) => {
+  const { config } = getQuestion(state, conditionQuestionId);
+  const {
+    condition: { conditions },
+  } = config;
+  const expressionParser = new BooleanExpressionParser();
+  const checkConditionMet = ({ formula, defaultValues }) => {
+    const values = {};
+    const variables = expressionParser.getVariables(formula);
+
+    variables.forEach(questionIdVariable => {
+      const questionId = questionIdVariable.replace(/^\$/, ''); // Remove the first $ prefix
+      const answer = getAnswerForQuestion(state, questionId);
+      const value = answer || defaultValues[questionId];
+      values[questionIdVariable] = value;
+    });
+
+    expressionParser.setScope(values);
+    return expressionParser.evaluate(formula);
+  };
+
+  const result = Object.keys(conditions).find(displayValue =>
```

Comment:
Are the keys of the conditions really a `displayValue`? I would have thought they were a question id...

That aside, I think we can use `Object.values` here

Reply:
The conditions config is something like:
```
conditions: {
    Yes: {
      formula: '$5fa0cdcd5da1e614f30001fd >= 3',
      defaultValues: {
         '5fa0cdcd5da1e614f30001fd': '0',
      },
    },
    No: {
      formula: '$5fa0cdcd5da1e614f30001fd < 3',
      defaultValues: {
        '5fa0cdcd5da1e614f30001fd': '0',
      },
    },
  }
```

so the keys are the values which would be generated if the associated condition is satisfied.